src/redux/reducers/highlightReducers.js

```diff
@@ -0,0 +1,44 @@
+import {
+  SET_HIGHLIGHTED_SECTION,
+  RESET_ARTICLE_HIGHLIGHT,
+  SET_HIGHLIGHTS
+} from "../actionTypes";
+
+const INITIAL_STATE = {
+  start: null,
+  end: null,
+  comment: "",
+  articleHighlights: {}
+};
+
+export default (state = INITIAL_STATE, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case SET_HIGHLIGHTED_SECTION:
+      return {
+        ...state,
+        ...payload
+      };
+    case RESET_ARTICLE_HIGHLIGHT:
+      return {
+        ...state,
+        ...INITIAL_STATE,
```

Comment:
Hi @abayo-luc, why INITIAL_STATE again? I thought because it's equal to state in function params its enough to use ...state?

Reply:
Not always will state equal to INITIAL_STATE.